openraft/src/core/admin.rs

```diff
@@ -132,27 +166,21 @@ impl<'a, D: AppData, R: AppDataResponse, N: RaftNetwork<D>, S: RaftStorage<D, R>
             return;
         }
 
-        let new_config;
+        let new_config_result;
 
-        let curr = &self.core.effective_membership.membership;
+        if stage == ChangeMembershipStage::Joint {
+            new_config_result = self.uniform_to_joint_config(members.borrow());
```

Comment:
The caller does not know about the current membership when sending ChangeMembership request to RaftCore.
At the first step changing uniform to joint config, it is possible it's already in an expected **joint** state.

E.g.:
- A client sent a change membership request;
- Leader committed it but failed to respond to the client.
- Client then re-send a change-membership request.

At this time, the leader can not just enter the joint state again.
But instead, the leader should check the current membership config and decide what to do.

It is the same with the second phase, if the membership is already the same as the config the client wants, it should return an Ok.

Thus the field `stage` can be just ignored.

Reply:
OK,get it,reset the commit